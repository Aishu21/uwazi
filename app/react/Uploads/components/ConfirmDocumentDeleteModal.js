import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import Modal from 'app/Layout/Modal';

import {hideModal} from 'app/Modals/actions/modalActions';
import {deleteDocument, finishEdit} from 'app/Uploads/actions/uploadsActions';

export class ConfirmDocumentDeleteModal extends Component {

  confirm() {
    this.props.finishEdit();
    this.props.hideModal('deleteDocument');
    this.props.deleteDocument(this.props.doc.toJS());
  }

  render() {
    if (!this.props.doc) {
      return <div />;
    }

    let doc = this.props.doc.toJS();

    return (
      <Modal isOpen={!!doc} type="danger">

        <Modal.Body>
          <h4>Confirm deletion</h4>
          <p>Are you sure you want to delete the document: {doc.title}</p>
        </Modal.Body>

        <Modal.Footer>
          <button type="button" className="btn btn-default cancel-button" onClick={() => this.props.hideModal('deleteDocument') }>
            <i className="fa fa-close"></i> Cancel
          </button>
          <button type="button" className="btn btn-danger confirm-button" onClick={() => this.confirm()}>
            <i className="fa fa-trash"></i> Delete document
          </button>
        </Modal.Footer>

      </Modal>
    );
  }
}

ConfirmDocumentDeleteModal.propTypes = {
  hideModal: PropTypes.func,
  deleteDocument: PropTypes.func,
  finishEdit: PropTypes.func,
  doc: PropTypes.object
};

const mapStateToProps = (state) => {
  return {doc: state.modals.get('deleteDocument')};
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({hideModal, deleteDocument, finishEdit}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmDocumentDeleteModal);
