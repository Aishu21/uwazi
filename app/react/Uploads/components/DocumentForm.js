import React, {Component, PropTypes} from 'react';
import {reduxForm} from 'redux-form';
import {bindActionCreators} from 'redux';

import Select from 'app/Form/components/Select';
import Textarea from 'app/Form/components/Textarea';
import {DynamicFields} from 'app/Form';
import {prepareTemplateFields, validate, generateValidation} from 'app/Form';
import {saveDocument, moveToLibrary} from 'app/Uploads/actions/uploadsActions';

export class DocumentForm extends Component {

  submit(e, move) {
    e.preventDefault();

    if (move) {
      return this.props.handleSubmit(this.props.moveToLibrary)();
    }

    this.props.handleSubmit(this.props.saveDocument)();
  }

  render() {
    const templateOptions = this.props.templates.map((template) => {
      return {label: template.name, value: template._id};
    });

    let fieldsTemplate = [];
    if (this.props.values.template) {
      let templateSelected = this.props.templates.find(template => template._id === this.props.values.template);
      fieldsTemplate = prepareTemplateFields(templateSelected, this.props.thesauris).properties;
    }

    return (
      <form onSubmit={(e) => this.submit(e, false)}>
        <Textarea properties={this.props.fields.title} label="Title" />
        <Select properties={this.props.fields.template} options={templateOptions} />
        <DynamicFields template={fieldsTemplate} fields={this.props.fields.metadata} />

        <button className="btn btn-sm btn-default submit">
          <i className="fa fa-floppy-o"></i>
          Save
        </button>
        {(() => {
          if (this.props.initialDoc.processed) {
            return (
              <button onClick={(e) => this.submit(e, true)} className="btn btn-sm btn-default to-library">
              <i className="fa fa-folder-open-o"></i>
              Move to library
              </button>
            );
          }
        })()}
      </form>
    );
  }

}

DocumentForm.propTypes = {
  fieldsConfig: PropTypes.array,
  templates: PropTypes.array,
  fields: PropTypes.object,
  initialDoc: PropTypes.object,
  values: PropTypes.object,
  thesauris: PropTypes.array,
  handleSubmit: PropTypes.func,
  saveDocument: PropTypes.func,
  moveToLibrary: PropTypes.func
};

export function mapStateToProps(state, props) {
  let currentTemplate = props.initialValues.template;

  if (state.form.document) {
    currentTemplate = state.form.document.template.value;
  }

  let template = props.templates.find((t) => t._id === currentTemplate);

  let metadataFields = template.properties.map((property) => {
    return 'metadata.' + property.name;
  });

  return {
    fieldsConfig: props.fieldsConfig,
    fields: ['_id', '_rev', 'title', 'template', ...metadataFields],
    template,
    initialDoc: props.initialValues
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({saveDocument, moveToLibrary}, dispatch);
}

export function validateForm(values, props) {
  let dynamicValidations = generateValidation(props.template, 'metadata.');
  return validate(values, Object.assign({
    title: {presence: true}
  }, dynamicValidations));
}

let form = reduxForm({
  form: 'document',
  validate: validateForm
},
mapStateToProps,
mapDispatchToProps
)(DocumentForm);

export default form;
