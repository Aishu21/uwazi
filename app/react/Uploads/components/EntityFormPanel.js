import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {saveEntity, finishEditEntity} from 'app/Uploads/actions/uploadsActions';
import SidePanel from 'app/Layout/SidePanel';
import EntityForm from '../containers/EntityForm';

export class EntityFormPanel extends Component {
  submit(entity) {
    this.props.saveEntity(entity);
  }

  render() {
    let sidePanelprops = {open: this.props.open};
    return (
      <SidePanel {...sidePanelprops}>
        <h1>Entity metadata</h1>
        <i className='fa fa-close close-modal' onClick={this.props.finishEditEntity}></i>
        <EntityForm onSubmit={this.submit.bind(this)}/>
      </SidePanel>
    );
  }
}

EntityFormPanel.propTypes = {
  open: PropTypes.bool,
  saveEntity: PropTypes.func,
  finishEditEntity: PropTypes.func,
  title: PropTypes.string
};

const mapStateToProps = ({uploads}) => {
  let uiState = uploads.uiState;
  return {
    open: uiState.get('showEntityForm'),
    title: uploads.entity.title
  };
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({saveEntity, finishEditEntity}, dispatch);
}
export default connect(mapStateToProps, mapDispatchToProps)(EntityFormPanel);
