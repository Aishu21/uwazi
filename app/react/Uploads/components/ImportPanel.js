import { connect } from 'react-redux';
import Immutable from 'immutable';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';

import { Translate } from 'app/I18N';
import SidePanel from 'app/Layout/SidePanel';
import { Icon } from 'UI';
import { LocalForm, Control } from 'react-redux-form';
import { closeImportPanel } from 'app/Uploads/actions/uploadsActions';


export class ImportPanel extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  close() {
    this.props.closeImportPanel();
  }

  handleSubmit(values) {
    console.log(values);
  }

  render() {
    return (
      <SidePanel open={this.props.open} className="metadata-sidepanel">
        <div className="sidepanel-header">
          <button className="closeSidepanel close-modal" onClick={this.close.bind(this)}>
            <Icon icon="times" />
          </button>
        </div>
        <LocalForm onSubmit={this.handleSubmit}>
          <div className="form-group">
            <ul className="search__filter">
              <label>Template</label>
              <Control.select model=".template" >
                {this.props.templates.map(t => <option value={t.get('_id')}>{t.get('name')}</option>)}
              </Control.select>
            </ul>
          </div>
          <div className="form-group" />
        </LocalForm>
      </SidePanel>
    );
  }
}

ImportPanel.defaultProps = {
  open: false,
};

ImportPanel.propTypes = {
  open: PropTypes.bool,
  templates: PropTypes.instanceOf(Immutable.List).isRequired,
  closeImportPanel: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
    open: state.importEntities.showImportPanel,
    templates: state.templates
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ closeImportPanel }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(ImportPanel);
