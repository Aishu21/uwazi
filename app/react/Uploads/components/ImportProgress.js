import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Translate } from 'app/I18N';
import { Icon } from 'UI';
import { bindActionCreators } from 'redux';
import { closeImportProgress } from 'app/Uploads/actions/uploadsActions';

export class ImportProgress extends Component {
  render() {
    const { close, importState: { importStart, importProgress, importError, importEnd } } = this.props;
    if (!importStart && !importProgress) {
      return false;
    }
    if (importEnd) {
      return (
        <div className="alert alert-info">
          <Icon icon="check" size="2x" />
          <div className="force-ltr">
            <Translate>Import completed</Translate>, {importProgress} <Translate>processed</Translate>
          </div>
          <Icon style={{ cursor: 'pointer' }} icon="times" onClick={close}/>
        </div>
      );
    }

    if (importError.get('message')) {
      return (
        <div className="alert alert-danger">
          <Icon icon="exclamation-triangle" size="2x" />
          <div className="force-ltr">
            <Translate>Import error</Translate>
            <p>{importError.get('message')}</p>
          </div>
          <Icon style={{ cursor: 'pointer' }} icon="times" onClick={close}/>
        </div>
      );
    }

    return (
      <div className="alert alert-info">
        <Icon icon="cog" size="2x" />
        <div className="force-ltr">
          <Translate>Importing data in progress</Translate>: {importProgress}
        </div>
      </div>
    );
  }
}

ImportProgress.propTypes = {
  importState: PropTypes.object.isRequired,
  close: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
    importState: state.importEntities
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ close: closeImportProgress }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(ImportProgress);
