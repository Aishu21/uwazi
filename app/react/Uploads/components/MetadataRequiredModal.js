import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import Modal from 'app/Layout/Modal';

import {hideModal} from 'app/Modals/actions/modalActions';
import {editDocument} from 'app/Uploads/actions/uploadsActions';
import {loadDocument} from 'app/DocumentForm/actions/actions';

export class MetadataRequiredModal extends Component {

  editDocument() {
    this.props.hideModal('metadataRequired');
    let doc = this.props.doc.toJS();
    this.props.loadDocument(doc, this.props.templates.toJS());
    this.props.editDocument(doc);
  }

  render() {
    if (!this.props.doc) {
      return <div />;
    }

    let doc = this.props.doc.toJS();
    return (
      <Modal isOpen={!!doc} type="warning">

        <Modal.Body>
          <p>The selected document is not complete:</p>
          <p><b>{doc.title}</b></p>
          <p>You need to fill the required metadata before you can publish it.</p>
        </Modal.Body>

        <Modal.Footer>
          <button type="button" className="btn btn-default cancel-button" onClick={() => this.props.hideModal('metadataRequired')}>
            <i className="fa fa-close"></i> Cancel
          </button>
          <button type="button" className="btn btn-warning confirm-button" onClick={this.editDocument.bind(this)}>
            <i className="fa fa-send"></i> Edit metadata
          </button>
        </Modal.Footer>

      </Modal>
    );
  }
}

MetadataRequiredModal.propTypes = {
  isOpen: PropTypes.bool,
  doc: PropTypes.object,
  templates: PropTypes.object,
  hideModal: PropTypes.func,
  editDocument: PropTypes.func,
  loadDocument: PropTypes.func
};

const mapStateToProps = (state) => {
  return {
    doc: state.modals.get('metadataRequired'),
    templates: state.uploads.templates
  };
};

function mapDispatchToProps(dispatch) {
  return bindActionCreators({hideModal, editDocument, loadDocument}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(MetadataRequiredModal);
