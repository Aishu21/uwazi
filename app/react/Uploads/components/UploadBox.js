import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import Dropzone from 'react-dropzone';

import {uploadDocument} from 'app/Uploads/actions/uploadsActions';
import 'app/Uploads/scss/upload_box.scss';

export class UploadBox extends Component {
  onDrop(files) {
    files.forEach((file) => {
      let doc = {title: this.extractTitle(file)};
      this.props.uploadDocument(doc, file);
    });
  }

  extractTitle(file) {
    let title = file.name
    .replace(/\.[^/.]+$/, '')
    .replace(/_/g, ' ')
    .replace(/-/g, ' ')
    .replace(/ {2}/g, ' ');

    return title.charAt(0).toUpperCase() + title.slice(1);
  }

  render() {
    return (
      <Dropzone style={{}} onDrop={this.onDrop.bind(this)}>
        <div className="upload-box">
          <i className="fa fa-upload"></i>
          <a>Upload your document</a>
        </div>
      </Dropzone>
    );
  }
}

UploadBox.propTypes = {
  uploadDocument: PropTypes.func
};


function mapDispatchToProps(dispatch) {
  return bindActionCreators({uploadDocument}, dispatch);
}

export default connect(null, mapDispatchToProps)(UploadBox);
