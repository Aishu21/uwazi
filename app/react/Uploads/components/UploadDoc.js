import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {RowList, ItemFooter, ItemName} from 'app/Layout/Lists';
import {editDocument, finishEdit} from 'app/Uploads/actions/uploadsActions';
import {loadDocument} from 'app/DocumentForm/actions/actions';
import {showModal} from 'app/Modals/actions/modalActions';

export class UploadDoc extends Component {
  showModal(modal, e) {
    if (modal) {
      e.stopPropagation();
      this.props.showModal(modal, this.props.doc);
    }
  }

  editDocument(doc, active) {
    if (active) {
      return this.props.finishEdit();
    }
    this.props.loadDocument(doc, this.props.templates.toJS());
    this.props.editDocument(doc);
  }

  render() {
    let doc = this.props.doc.toJS();
    let modal = false;

    let status = 'success';
    let message = 'Ready for publish';

    if (!doc.template) {
      status = 'warning';
      message = 'Metadata required';
      modal = 'metadataRequired';
    }

    if (doc.uploaded === false) {
      status = 'danger';
      message = 'Upload failed';
      modal = '';
    }

    if (!doc.processed && doc.uploaded) {
      status = 'info';
      message = 'Processing...';
      modal = '';
    }

    if (doc.processed === false) {
      status = 'danger';
      message = 'Conversion failed';
      modal = '';
    }

    let itsUploading = typeof this.props.progress === 'number';

    if (itsUploading) {
      status = 'info';
      modal = '';
    }

    let active;
    if (this.props.documentBeingEdited) {
      active = this.props.documentBeingEdited === doc._id;
    }

    return (
      <RowList.Item status={status} active={active} onClick={this.editDocument.bind(this, doc, active)}>
      <ItemName>{doc.title}</ItemName>
      <ItemFooter onClick={this.showModal.bind(this, modal)}>
        {(() => {
          if (itsUploading) {
            return <ItemFooter.ProgressBar progress={this.props.progress} />;
          }
          return <ItemFooter.Label status={status}>{message}</ItemFooter.Label>;
        })()}
      </ItemFooter>
    </RowList.Item>
    );
  }
}

UploadDoc.propTypes = {
  doc: PropTypes.object,
  progress: PropTypes.number,
  editDocument: PropTypes.func,
  documentBeingEdited: PropTypes.string,
  loadDocument: PropTypes.func,
  finishEdit: PropTypes.func,
  showModal: PropTypes.func,
  templates: PropTypes.object
};

export function mapStateToProps(state, props) {
  return {
    progress: state.uploads.progress.get(props.doc.get('_id')),
    documentBeingEdited: state.uploads.uiState.get('documentBeingEdited'),
    templates: state.uploads.templates
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({finishEdit, editDocument, loadDocument, showModal}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(UploadDoc);
