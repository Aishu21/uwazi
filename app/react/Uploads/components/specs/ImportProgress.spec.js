import React from 'react';

import { ImportProgress } from 'app/Uploads/components/ImportProgress';
import { shallow } from 'enzyme';

describe('ImportProgress', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      importState: { importStart: true, importProgress: 5, importError: '', importEnd: false },
      close: jasmine.createSpy('close'),
    };
  });

  const render = () => {
    component = shallow(<ImportProgress {...props}/>);
  };

  describe('rendering states', () => {
    it('should render a state for normal progress', () => {
      render();
      expect(component).toMatchSnapshot();
    });

    it('should render a state for errors', () => {
      props.importState.importError = 'Something bad happened';
      render();
      expect(component).toMatchSnapshot();
    });

    it('should render a state for end', () => {
      props.importState.importEnd = true;
      render();
      expect(component).toMatchSnapshot();
    });
  });
});
