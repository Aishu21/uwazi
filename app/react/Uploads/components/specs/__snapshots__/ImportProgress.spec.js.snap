// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportProgress rendering states should render a state for end 1`] = `
<div
  className="alert alert-info"
>
  <Connect(Icon)
    icon="check"
    size="2x"
  />
  <div
    className="force-ltr"
  >
    <Connect(Translate)>
      Import completed
    </Connect(Translate)>
    , 
    5
     
    <Connect(Translate)>
      processed
    </Connect(Translate)>
  </div>
  <Connect(Icon)
    icon="times"
    style={
      Object {
        "cursor": "pointer",
      }
    }
  />
</div>
`;

exports[`ImportProgress rendering states should render a state for errors 1`] = `
<div
  className="alert alert-danger"
>
  <Connect(Icon)
    icon="exclamation-triangle"
    size="2x"
  />
  <div
    className="force-ltr"
  >
    <Connect(Translate)>
      Import error
    </Connect(Translate)>
    : 
    Something bad happened
  </div>
  <Connect(Icon)
    icon="times"
    style={
      Object {
        "cursor": "pointer",
      }
    }
  />
</div>
`;

exports[`ImportProgress rendering states should render a state for normal progress 1`] = `
<div
  className="alert alert-info"
>
  <Connect(Icon)
    icon="cog"
    size="2x"
  />
  <div
    className="force-ltr"
  >
    <Connect(Translate)>
      Importing data in progress
    </Connect(Translate)>
    : 
    5
  </div>
</div>
`;
